test/csuite/Makefile.am

```diff
@@ -57,6 +57,9 @@ noinst_PROGRAMS += test_wt3135_search_near_collator
 test_wt3184_dup_index_collator_SOURCES = wt3184_dup_index_collator/main.c
 noinst_PROGRAMS += test_wt3184_dup_index_collator
 
+test_wt3338_partial_update_SOURCES = wt3338_partial_update/main.c
+noinst_PROGRAMS += test_wt3338_partial_update
+
```

Comment:
> and nothing specifically tests modify with logging enabled that I can tell.

`test/format` tests modify with logging enabled.

> Also, it would be good to have a new test section in test_compat01.py or even a new test_compat02.py that tests modify and compatibility, confirming that the new record never ends up in the old version log file.

That would probably have to wait on the resolution of WT-3417.

Should we instead add upgrade/downgrade support to `test/format`?